src/joint_constraints.rs

```diff
@@ -287,32 +289,29 @@ pub fn propagate_joint_to_var_bounds(vcs: &mut VarConstraints, jcs: &JointConstr
             let sum_other_min: usize = jc.vars
                 .iter()
                 .filter(|&&w| w != v)
-                .map(|&w| vcs.bounds(w).0)
+                .map(|&w| vcs.bounds(w).0.unwrap_or(VarConstraint::DEFAULT_MIN))
                 .sum();
 
             // Σ other finite maxes (None if any is ∞)
             let mut sum_other_max_opt: Option<usize> = Some(0);
             for &w in jc.vars.iter().filter(|&&w| w != v) {
                 let (_, w_ui) = vcs.bounds(w);
-                if w_ui == VarConstraint::DEFAULT_MAX {
+                if w_ui.is_none() {
                     sum_other_max_opt = None;
                     break;
                 }
-                sum_other_max_opt = sum_other_max_opt.map(|a| a + w_ui);
+                sum_other_max_opt = sum_other_max_opt.and_then(|a| w_ui.map(|w| a + w));
             }
 
-            let lower_from_joint = match sum_other_max_opt {
-                Some(s) => jc.target.saturating_sub(s),
-                None => 0, // others can stretch arbitrarily
-            };
+            let lower_from_joint = sum_other_max_opt.map(|s| jc.target.saturating_sub(s));
             let upper_from_joint = jc.target.saturating_sub(sum_other_min);
 
             // Tighten and store
             let new_min = li.max(lower_from_joint);
-            let new_max = ui.min(upper_from_joint);
+            let new_max = ui.map_or(upper_from_joint, |u| u.min(upper_from_joint)); // TODO is there a better way to write this (NB: min(None, x) is always None (I think...))
```

Comment:
```suggestion
            let new_max = ui.unwrap_or(upper_from_joint).min(upper_from_joint);
```
if you want, otherwise what you have is fine i think